src/http/common.cc

```diff
@@ -124,6 +124,48 @@ output_stream<char> make_http_chunked_output_stream(output_stream<char>& out) {
     return output_stream<char>(http_chunked_data_sink(out), 32000, opts);
 }
 
+class http_content_length_data_sink_impl : public data_sink_impl {
+    output_stream<char>& _out;
+    const size_t _limit;
+    size_t& _bytes_written;
+
+public:
+    http_content_length_data_sink_impl(output_stream<char>& out, size_t& len)
+            : _out(out)
+            , _limit(std::exchange(len, 0))
+            , _bytes_written(len)
+    {
+    }
+    virtual future<> put(net::packet data)  override { abort(); }
+    using data_sink_impl::put;
+    virtual future<> put(temporary_buffer<char> buf) override {
+        if (buf.size() == 0 || _bytes_written == _limit) {
+            return make_ready_future<>();
+        }
+        auto size = std::min(buf.size(), _limit - _bytes_written);
+        return _out.write(buf.get(), size).then([this, size] {
+            _bytes_written += size;
+        });
+    }
+    virtual future<> close() override {
+        return make_ready_future<>();
+    }
+};
+
+class http_content_length_data_sink : public data_sink {
+public:
+    http_content_length_data_sink(output_stream<char>& out, size_t& len)
+        : data_sink(std::make_unique<http_content_length_data_sink_impl>(out, len))
+    {
+    }
+};
+
+output_stream<char> make_http_content_length_output_stream(output_stream<char>& out, size_t& len) {
+    output_stream_options opts;
+    opts.trim_to_size = true;
+    return output_stream<char>(http_content_length_data_sink(out, len), 32000, opts);
```

Comment:
It is unclear what 32000 stands for.
I suggest defining it as a named constexpr.
It would also be nice to understand why it's set to 32000 and not to e.g. 32768?

Reply:
> It is unclear what 32000 stands for.

buffer_size. It's the same as in pre-existing `make_http_chunked_output_stream()`

> I suggest defining it as a named constexpr.
> It would also be nice to understand why it's set to 32000 and not to e.g. 32768?

https://www.reddit.com/r/space/comments/k4x1gq/the_connection_between_horses_asses_and_space/ :D